awx/main/scheduler/dag_workflow.py

```diff
@@ -215,10 +214,19 @@ def mark_dnr_nodes(self):
             obj = node['node_object']
 
             if obj.do_not_run is False and not obj.job and node not in root_nodes:
-                parent_nodes = [p['node_object'] for p in self.get_dependents(obj)]
+                parent_nodes = [p['node_object'] for p in self.get_parents(obj)]
                 if self._are_all_nodes_dnr_decided(parent_nodes):
                     if self._should_mark_node_dnr(node, parent_nodes):
                         obj.do_not_run = True
                         nodes_marked_do_not_run.append(node)
+            if obj.all_parents_must_converge:
```

Comment:
maaaybe not necessary, but I feel like this should check `if (not obj.do_not_run) and obj.all_parents_must_converge`, otherwise I could imagine duplicate entries in `nodes_marked_do_not_run` because it's a list and not a set

Reply:
That is fair, gives it less list to process, I'll add that.